R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+#' paste3
+#'
+#' An internal fucntion that removes NAs from paste.
+#' Source: https://stackoverflow.com/users/1855677/42.
+#'
+#' @param ... objects to be pasted together.
+#' @param sep separating string.
+
+paste3 <- function(..., sep = ", ") {
+  L <- list(...)
+  L <- lapply(L, function(x) {
+    x[is.na(x)] <- ""
+    x
+  })
+  ret <- gsub(
+    paste0("(^", sep, "|", sep, "$)"), "",
+    gsub(
+      paste0(sep, sep), sep,
+      do.call(paste, c(L, list(sep = sep)))
+    )
+  )
+  is.na(ret) <- ret == ""
+  ret
+}
+
+#' Supply an expression to identify which header groups are hooked.
+#'
+#' @description
+#' This function is used with the `.hook_if` or `.hook_if_rev` arguments in the `locate_groups` function.
+#' It passes an expression to `dplyr::summarise` that identifies which header groups are hooked - for example swiched from N to NNW.
+#' See the the `locate_groups` documentation for more information and an example.
+#'
+#' @param ...  expression applied to a  identifies which header groups are hooked
+#'
+#' @name get_range_dfs
+#' @export
+
+hook <- function(...) {
+  rlang::quos(...)
+}
+
+#' Unmerge and fill in blanks for row headers.
+#'
+#' This is an internal function that function ensures that merged cells are unmerged.
+#'
+#' @param header_df a data frame containing header cells.
+#' @param header_fill the criteria according to which blank cells will be filled.
+#' @param formats the formats of the workbook associated with the tidyxl data frame.
+
+fill_blanks_in_row_headers <- function(header_df, header_fill, formats) {
```

Comment:
It would be good to control the inputs to `header_fill`, either by `header_fill = c("style", "local_format_id", "borders")` followed by `header_fill = match.arg(header_fill)`, or `stopifnot(header_fill %in% c("style", "local_format_id", "borders")`

Reply:
Agreed. I'll put this through.

Reply:
Addressed in 93c4d07929a0ae36947e9a4f335a39dfe8382d9e.